ledger/store/interface.go

```diff
@@ -133,3 +171,26 @@ type CatchpointReaderWriter interface {
 	CatchpointReader
 	CatchpointWriter
 }
+
+// StateProofReader is a reader abstraction for stateproof verification tracker
+// Use with SnapshotScope
+type StateProofReader interface {
```

Comment:
the StateProofReader might be confusing (might hint that we read and write state proofs which is not the case). Should consider using the words `verification` and `context`. maybe: `spVerificatinCtxReader`